test/test_nn.py

```diff
@@ -9726,20 +9726,35 @@ def test_embedding_padding_idx(self, device, dtype):
 
     @dtypesIfCUDA(torch.half, torch.float)
     @dtypes(torch.float)
-    def test_softmax_backward(self, device, dtype):
+    def test_softmax_results(self, device, dtype):
         # Non-even sizes and non-zero shifts test fallback paths in vectorized kernel
         # Note: dim1 > 1024 is needed to exercise the vectorized (non-persistent) path, (16, 30576) is BERT-esque
         sizes = [(0, 10), (32, 20), (10, 0), (31, 20), (32, 21), (31, 23), (32, 1536), (31, 2048), (16, 30576)]
         shifts = [(0, 0), (1, 0), (0, 1), (1, 1)]
         for fn in [F.softmax, F.log_softmax]:
             for size in sizes:
                 for shift in shifts:
-                    input = torch.rand(size, device=device, dtype=dtype, requires_grad=True)
+                    input = torch.rand(size, device=device, dtype=dtype)
+                    # Note: With the largest tests we can hit upper limit of fp16 when we
+                    # sum, so scale the input down to stay in a nicer range.
+                    if dtype == torch.float16:
+                        input = input / 100.
                     input = input[shift[0]:, shift[1]:]
+                    # Note; Don't want to bprop back through slice op
+                    input = input.detach().requires_grad_(True)
+                    ref_input = input.clone().cpu().detach().requires_grad_(True)
                     for dim in [0, 1]:
+                        ref_output = fn(ref_input, dtype=torch.float, dim=dim).sum()
```

Comment:
nit: when you calculate output as .sum(), gradOutput will be filled with ones, which won't let you detect if you are reading gradOutput values from the wrong places. You can do
``` 
output=fn(input)
gradOutput = torch.rand_like(output)
grad_input, = torch.autograd.grad(output, input, grad_outputs=(gradOutput), create_graph=True)
```

Reply:
Is that style necessary or is this syntax acceptable?
```
grad_output = torch.rand_like(output)
output.backward(grad_output)
```

Reply:
Ended up using your syntax (although I'm still interested in _why_ that syntax over the simpler one I mentioned)